pkg/assets/file_builder.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package assets
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"k8s.io/kops/pkg/apis/kops"
+)
+
+// AssetBuilder discovers and remaps assets
+type FileAssetBuilder struct {
```

Comment:
I think you could incorporate this into AssetBuilder, rather than needing a separate class, but we can see how it plays out..